### PR TITLE
Allow calc_analysis.x to switch between JEDI and GSI-formatted increments in calc_analysis.nml

### PR DESCRIPTION
**Pull request by DavidNew-NOAA**
Because the JEDI increments will soon be produced in the Global Workflow with a time dimension, calc_analysis.x will fail, since it expects a GSI-style increment with no time dimensions. The PR adds a "jedi" parameter in calc_analysis.nml that is default set to false. When it is false, it expects no time dimension when reading and adding the increment to the background, and when it is true, it does expect a time dimension. 

I have tested this in the Global Workflow by adding and removing the time dimension for the atmospheric increment, and running calc_analysis.x in the "gdasanalcalc" job with jedi set to .true. and not set at all (defaults to false) respectively. The resulting analyses are zero diff.

### CODE REVIEW

File: src/netcdf_io/calc_analysis.fd/inc2anl.f90 [hunk 1]
```diff
@@ -196,12 +197,22 @@ subroutine add_increment(fcstvar, incvar)
...
           ! get increment
           incncfile = open_dataset(incr_file, paropen=.true.)
-          call read_vardata(incncfile, trim(incvar)//"_inc", work3d_inc, nslice=k, slicedim=3)
-          ! add increment to background
-          do j=1,nlat
-             jj=nlat+1-j ! increment is S->N, history files are N->S
-             work3d_bg(:,j,1) = work3d_bg(:,j,1) + work3d_inc(:,jj)
-          end do
+          ! JEDI-derived increments have a time dimension, so read to appropriate array
+          if ( jedi ) then
```
Comment (CoryMartin-NOAA) [inline src/netcdf_io/calc_analysis.fd/inc2anl.f90:201]: This is great. I suspect when we move from the AuxGrid in FV3-JEDI to the generic Gaussian interpolation/write in OOPS, this boolean will be even more important.

**Comment by CoryMartin-NOAA** [comment]
@DavidNew-NOAA can you (or have you already) run a low resolution 3DVar test using develop of g-w with this hash of gsi_utils just to confirm that it will cycle with GSI still? It should be fine but I would feel better testing it to be sure.

**Comment by DavidNew-NOAA** [comment]
@CoryMartin-NOAA I just ran this branch in GSI with GW feature/jediinc2fv3 (will be merged into develop soon, but test should work in develop already), and it got through a full cycle. It's moving through the next cycle now without problem, including gdasanalcalc of that cycle.

**Comment by CoryMartin-NOAA** [comment]
Thanks @DavidNew-NOAA works for me!

**Review by CoryMartin-NOAA — APPROVED** [review]

**Pull request merged**